pandas/tests/scalar/timedelta/test_timedelta.py

```diff
@@ -918,6 +920,35 @@ def test_timedelta_hash_equality(self):
         ns_td = Timedelta(1, "ns")
         assert hash(ns_td) != hash(ns_td.to_pytimedelta())
 
+    @pytest.mark.xfail(
+        reason="pd.Timedelta violates the Python hash invariant (GH#44504).",
+        raises=AssertionError,
+        strict=True,
+    )
+    @given(
+        st.integers(
+            min_value=(-sys.maxsize - 1) // 500,
+            max_value=sys.maxsize // 500,
+        )
+    )
+    def test_hash_equality_invariance(self, half_microseconds: int) -> None:
+        # GH#44504
+
+        def _upholds_hash_equality_invariance(v1: Hashable, v2: Hashable, /) -> bool:
```

Comment:
This helper function doesn't seem necessary since it's only called once

Reply:
I didn't create this function with reuse in mind. The intention behind the function is to be able to label a piece of code with a name, in this case `_upholds_hash_equality_invariance`, as well as provide some type annotation for it.

I see this as a preferable approach to documentation compared with writing comments, which aren't statically checked, and are rarely kept up to date.

I can take the other approach if you feel strongly that is's better, though.

Reply:
We don't use type annotations in the test suite, and I think the test name plus the comments should be clear enough to inform the test. I find this harder to understand due to the indirection of a function that is used once.

Reply:
I think you're missing out when not using annotations in the test suite. I often find typing helpful in the testing context. It makes helper functions, like this one, clearer. Also, providing types for test parameters makes up for the opaque way fixtures are fed to test methods, helping with IDE type inference and autocomplete.

That said, I removed the function for you.